[GStreamer][Quirks] Fix playback rate change

eocanha opened the pull request:
#### 0788b53042c53ccce2c77b5c26a91db0d8413f81
<pre>
[GStreamer][Quirks] Fix playback rate change
<a href="https://bugs.webkit.org/show_bug.cgi?id=294438">https://bugs.webkit.org/show_bug.cgi?id=294438</a>

Reviewed by Xabier Rodriguez-Calvar.

When attempting to change playback rate while the pipeline is not
playing and not prerolling (isPipelinePlaying = 0 and
isPipelineWaitingPreroll = 0), if the quirk processes the request but
doesn&apos;t change the playback rate, the rate won&apos;t get changed on
platforms where instant playback rate change is supported.

See: <a href="https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1515">https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1515</a>

This patch resorts to the regular seek based approach used in platforms
with no quirks when the instant rate change fails on a platform with
quirks needing instant rate change.

Original author: Filipe Norte &lt;filipe_norte@comcast.com&gt;

* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::updatePlaybackRate): Resort to doSeek() when the instant rate change couldn&apos;t be completed.

Canonical link: <a href="https://commits.webkit.org/296191@main">https://commits.webkit.org/296191@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3898cee50a8fcf2d7aba91b872ffe9c4ef90ac33

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107658 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27334 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17748 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112870 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58198 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109615 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28026 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35851 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81739 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110591 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22213 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97031 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62119 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21647 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15171 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57632 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91585 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15204 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/115988 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34734 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/25604 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90782 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35111 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93281 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90528 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23082 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35446 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13218 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30491 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34640 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/40196 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34386 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37747 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36049 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3898cee50a8fcf2d7aba91b872ffe9c4ef90ac33. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46708

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3898cee50a8fcf2d7aba91b872ffe9c4ef90ac33. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46708

eocanha committed:
[GStreamer][Quirks] Fix playback rate change
https://bugs.webkit.org/show_bug.cgi?id=294438

Reviewed by Xabier Rodriguez-Calvar.

When attempting to change playback rate while the pipeline is not
playing and not prerolling (isPipelinePlaying = 0 and
isPipelineWaitingPreroll = 0), if the quirk processes the request but
doesn't change the playback rate, the rate won't get changed on
platforms where instant playback rate change is supported.

See: https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1515

This patch resorts to the regular seek based approach used in platforms
with no quirks when the instant rate change fails on a platform with
quirks needing instant rate change.

Original author: Filipe Norte <filipe_norte@comcast.com>

* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::updatePlaybackRate): Resort to doSeek() when the instant rate change couldn't be completed.

Canonical link: https://commits.webkit.org/296191@main
